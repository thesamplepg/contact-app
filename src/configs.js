export const inputs = {
  name: {
    type: "text",
    value: "",
    required: 3
  },
  address: {
    type: "text",
    value: "",
    required: 6
  },
  phone: {
    type: "number",
    value: "",
    required: 9
  }
};
