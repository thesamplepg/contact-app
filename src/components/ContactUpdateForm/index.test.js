import React from "react";
import { shallow } from "enzyme";

import Component from "./index";
import { inputs } from "../../configs";

describe("ContactUpdateForm component", () => {
  const props = {
    inputs,
    errors: []
  };

  const renderComponent = props => shallow(<Component {...props} />);

  it("render propertly", () => {
    expect(renderComponent(props)).toMatchSnapshot();
  });

  it("should render all inputs", () => {
    const component = renderComponent(props);
    expect(component.find("input")).toHaveLength(3);
  });

  it("input[name] should have a class <error>", () => {
    const nextProps = { ...props, errors: ["name"] };
    const component = renderComponent(nextProps);

    const nameInput = component.find('input[id="name"]');

    expect(nameInput.hasClass("error")).toBeTruthy();
  });
});
