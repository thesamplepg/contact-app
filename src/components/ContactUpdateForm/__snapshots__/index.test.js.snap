// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactUpdateForm component render propertly 1`] = `
<li
  className="UpdateItem"
>
  <input
    className={null}
    id="name"
    key="name"
    placeholder="name"
    type="text"
    value=""
  />
  <input
    className={null}
    id="address"
    key="address"
    placeholder="address"
    type="text"
    value=""
  />
  <input
    className={null}
    id="phone"
    key="phone"
    placeholder="phone"
    type="number"
    value=""
  />
  <button>
    update
  </button>
</li>
`;
